### PR TITLE
chore: Enable Typescript strict mode

### LINKED ISSUE #26891: [Feature]: enable TS strict mode in wc3
### Library

Web Components (@fluentui/web-components)

### Describe the feature that you would like added

ATM web-components use very weak TS compiler checks .

This introduces various issues in terms of compatibility with other packages in monorepo and scaleability of the library.

**Current:**

```
"strictNullChecks": true,
```

**Expected:**

```
"strict": true,
```

### Have you discussed this feature with our team

@miroslavstastny 

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by miroslavstastny**
Enables Typescript strict mode for Web Components v3.

Fixes #26891

### CODE REVIEW

File: packages/web-components/src/counter-badge/counter-badge.template.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { ElementViewTemplate, html } from '@microsoft/fast-element';
 import { badgeTemplate } from '../badge/badge.template.js';
+import { Badge } from '../badge/badge.js';
 import { CounterBadge } from './counter-badge.js';
 import { CounterBadgeOptions } from './counter-badge.options.js';
 
-function composeTemplate<T extends CounterBadge>(options: CounterBadgeOptions = {}): ElementViewTemplate<T> {
-  return badgeTemplate({
+function composeTemplate<T extends CounterBadge & Badge>(options: CounterBadgeOptions = {}): ElementViewTemplate<T> {
```
Comment (miroslavstastny) [inline packages/web-components/src/counter-badge/counter-badge.template.ts:7]: `CounterBadge` composes `Badge` template but does not inherit from it.
@chrisdholt Are you OK with this change or would you propose a different fix?

**Review by micahgodbolt — APPROVED** [review]

File: packages/web-components/src/counter-badge/counter-badge.template.ts (hunk 1 above)
Comment (chrisdholt) [reply packages/web-components/src/counter-badge/counter-badge.template.ts:7]: Let's leave it for now and see how it goes

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**